### PR TITLE
[2/5] Create registration dialog shell

### PR DESCRIPTION
**Pull request by brittanylau**
### Notion ticket link

[Registration form modal shell](https://www.notion.so/uwblueprintexecs/9cb454262ed84c6299c0682c23f92ec5?v=1076bd30711a4a01838becee05f8b5a3&p=e0143b8253524334b986f573d3eaa375)

### Implementation description

- created a new `RegistrationDialog` component that's opened with the "new registrant" button on the sessions page
- created a `RegistrationForm` stub component that will contain the form contents
- within the dialog, if users click "Register a new client", the form will appear
- fixed some styling stuff

### Steps to test

1. `yarn start`
2. go to /sessions and click "New registrant" - the dialog should open
3. click "Register a new client" - some text showing "Currently enrolling a new family" should appear
4. click "Go back" - this should hide the text, and show the "Register a new client" button again

### What should reviewers focus on?

- testing!
- good react practices

### Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] I have run the linter
- [x] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: src/components/registration/RegistrationDialog.jsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  const classes = useStyles();
+  const [displayForm, setDisplayForm] = useState(false);
+
+  const handleDisplayForm = () => {
+    setDisplayForm(true);
+  };
+
+  const handleHideForm = () => {
+    setDisplayForm(false);
+  };
```
Comment (Tayef-Shah) [inline src/components/registration/RegistrationDialog.jsx:33]: are having these helper functions considered better practice than just explicitly calling `setDisplayForm()`?

File: src/pages/Sessions.jsx [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
-import { Typography } from "@material-ui/core";
+import { Button, Typography } from "@material-ui/core";
+import { Add } from "@material-ui/icons";
 
 import SessionAPI from "../api/SessionAPI";
+import RegistrationDialog from "../components/registration/RegistrationDialog";
 
 function Sessions() {
   const [sessions, setSessions] = useState([]);
+  const [displayRegDialog, setDisplayRegDialog] = React.useState(false);
```
Comment (Tayef-Shah) [inline src/pages/Sessions.jsx:10]: why are we explicitly calling useState from React here, but not on the line above?

File: src/pages/Sessions.jsx [hunk 3]
```diff
@@ -13,9 +16,25 @@ function Sessions() {
...
+  const handleOpenFormDialog = () => {
+    setDisplayRegDialog(true);
+  };
+
+  const handleCloseFormDialog = () => {
+    setDisplayRegDialog(false);
+  };
+
   return (
     <>
```
Comment (Tayef-Shah) [inline src/pages/Sessions.jsx:28]: is this supposed to add spacing? Or non-intentional?

**Review by Tayef-Shah — APPROVED** [review]
Look good to me :) I just commented on some minor things

File: src/pages/Sessions.jsx (hunk 2 above)
Comment (brittanylau) [reply src/pages/Sessions.jsx]: oof good catch, updated!

File: src/pages/Sessions.jsx (hunk 3 above)
Comment (brittanylau) [reply src/pages/Sessions.jsx:28]: this is a [fragment](https://reactjs.org/docs/fragments.html)! it lets us return multiple elements in a single component, without adding things to the DOM

we could've used a div too, but I don't think we need it for anything

**Pull request merged**